src/huggingface_hub/commit_api.py

```diff
@@ -0,0 +1,444 @@
+import base64
+import io
+import os
+from dataclasses import dataclass
+from functools import partial
+from hashlib import sha256
+from math import ceil
+from os.path import getsize
+from typing import Dict, Generator, Iterable, List, Literal, Optional, TypedDict, Union
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+from .constants import ENDPOINT, REPO_TYPES_URL_PREFIXES
+from .utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+UploadMode = Literal["lfs", "regular"]
+
+
+@dataclass
+class CommitOperationDelete:
+    path_in_repo: str
+
+
+@dataclass
+class CommitOperationAdd:
+    path_in_repo: str
+    path_or_fileobj: Union[str, bytes, io.BufferedIOBase]
+
+    def __post_init__(self):
+        if isinstance(self.path_or_fileobj, str):
+            path_or_fileobj = os.path.normpath(os.path.expanduser(self.path_or_fileobj))
+            if not os.path.isfile(path_or_fileobj):
+                raise ValueError(f"Provided path: '{path_or_fileobj}' is not a file")
+        elif not isinstance(self.path_or_fileobj, (io.BufferedIOBase, bytes)):
+            # ^^ Test from: https://stackoverflow.com/questions/44584829/how-to-determine-if-file-is-opened-in-binary-or-text-mode
+            raise ValueError(
+                "path_or_fileobj must be either an instance of str, bytes or BinaryIO."
+                " If you passed a file-like object, make sure it is in binary mode."
+            )
+
+    def iter_content(
+        self, chunk_size: Optional[int] = None
+    ) -> Generator[bytes, None, None]:
+        if isinstance(self.path_or_fileobj, bytes):
+            if chunk_size is None:
+                yield self.path_or_fileobj
+            else:
+                idx = 0
+                while chunk := self.path_or_fileobj[idx : idx + chunk_size]:
```

Comment:
Walrus operator has been introduced in Python 3.8

Will need to rewrite this function to be compatible with 3.7